src/platform/pico/uart_channel.c

```diff
@@ -0,0 +1,240 @@
+#include "reactor-uc/platform/pico/uart_channel.h"
+#include "reactor-uc/platform/pico/pico.h"
+#include "reactor-uc/logging.h"
+#include "reactor-uc/serialization.h"
+
+#define UART_CHANNEL_ERR(fmt, ...) LF_ERR(NET, "UartPolledChannel: " fmt, ##__VA_ARGS__)
+#define UART_CHANNEL_WARN(fmt, ...) LF_WARN(NET, "UartPolledChannel: " fmt, ##__VA_ARGS__)
+#define UART_CHANNEL_INFO(fmt, ...) LF_INFO(NET, "UartPolledChannel: " fmt, ##__VA_ARGS__)
+#define UART_CHANNEL_DEBUG(fmt, ...) LF_DEBUG(NET, "UartPolledChannel: " fmt, ##__VA_ARGS__)
+
+extern Environment *_lf_environment;
+
+UartPolledChannel *uart_channel_0 = NULL;
+UartPolledChannel *uart_channel_1 = NULL;
+FederateMessage msg;
+
+static lf_ret_t UartPolledChannel_open_connection(NetworkChannel *untyped_self) {
+  UartPolledChannel *self = (UartPolledChannel *)untyped_self;
+  char connect_message[] = {0x7, 0x7, 0x7, 0x7};
+  uart_write_blocking(self->uart_device, (const uint8_t *)connect_message, 4);
+  UART_CHANNEL_DEBUG("Open connection");
+  return LF_OK;
+}
+
+static void UartPolledChannel_close_connection(NetworkChannel *untyped_self) {
+  UART_CHANNEL_DEBUG("Close connection");
+  (void)untyped_self;
+}
+
+static void UartPolledChannel_free(NetworkChannel *untyped_self) {
+  UART_CHANNEL_DEBUG("Free");
+  (void)untyped_self;
+}
+
+static bool UartPolledChannel_is_connected(NetworkChannel *untyped_self) {
```

Comment:
This one just sleeps for 500 msec before saying that we are connected?

Reply:
This function was intended to check for a connection and possibly do some work towards establishing the connection and returning. We expect the runtime to poll this function for all channels, one after the other, and then sleeping, until they all return true